langtest/augmentation/__init__.py

```diff
@@ -241,3 +248,178 @@ def _parameters_overrides(self, config: dict, data_handler: List[Sample]) -> dic
+        templates (Union[str, List[str]]): The templates to be used for the augmentation.
+        task (str): The task for which the augmentation is being performed.
+        """
+        self.__templates: Union[str, List[str], List[Sample]] = templates
+        self.__task = task
+
+        if isinstance(self.__templates, str) and os.path.exists(self.__templates):
+            self.__templates = DataFactory(self.__templates, self.__task).load()
+        elif isinstance(self.__templates, str):
+            self.__templates = [self.str_to_sample(self.__templates)]
+        elif isinstance(self.__templates, list):
+            self.__templates = [self.str_to_sample(i) for i in self.__templates]
+
+    def fix(self, input_path: str, output_path: str, min_samples=5000, *args, **kwargs):
+        """
+        This method is used to perform the templatic augmentation.
+        It takes the input data, performs the augmentation and then saves the augmented data to the output path.
+
+        Parameters:
+        input_path (str): The path to the input data.
+        output_path (str): The path where the augmented data will be saved.
+        *args: Variable length argument list.
+        **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+        bool: Returns True upon successful completion of the method.
+        """
+
+        df = DataFactory(input_path, self.__task)
+        data = df.load()
+        new_data = []
+        self.__search_results = self.search_sample_results(data)
+
+        for template in self.__templates:
+            for _ in range(min_samples):
+                new_sample = self.new_sample(template)
+                if new_sample:
+                    new_data.append(new_sample)
+
+        df.export(new_data, output_path)
+
+        return True
+
+    def search_sample_results(
+        self, samples: List[Sample]
+    ) -> Dict[str, List[Union[NERPrediction, SequenceLabel]]]:
+        """"""
+        results_dict = defaultdict(list)
+        for sample in samples:
+            chunk = []
+            ent_name = ""
+            for result in sample.expected_results.predictions:
+                ent = result.entity.split("-")[-1]
+                if ent != "O" and ent_name == "":
+                    ent_name = ent
+                if result.entity.endswith(ent_name) and ent != "O":
+                    chunk.append(result)
+                elif len(chunk) > 0:
+                    results_dict[ent_name].append(tuple(chunk))
+                    ent_name = ""
+                    chunk = []
+
+            if chunk:
+                results_dict[ent_name].append(tuple(chunk))
+        return results_dict
+
+    def extract_variable_names(self, f_string: str):
+        pattern = r"{([^{}]*)}"
+        matches = re.findall(pattern, f_string)
+        variable_names = [match.strip() for match in matches]
+        return variable_names
+
+    def new_sample(self, template: Sample):
+        template = copy(template)
```

Comment:
I think we could drastically simplify the logic by using string formatting and keyword arguments.. Something like
```
s = f"My name is {PERS}"
s.format(PERS="John")
```

Reply:
yes, the format option is suitable for modifying the original in the sample, but how can we update the **expected_results.predictions** while swapping?

after augmentation, Conll file is generated based on **expected_results.predictions**

Reply:
For now I'm not getting how we properly compute the predictions spans.. Can you please add tests so that I can follow the process flow?

Reply:
we don't need span.start and span.end values while in augmentation.